.github/workflows/publish.yml

```diff
@@ -0,0 +1,51 @@
+name: publish
+
+on:
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  tests:
+    uses: ./.github/workflows/test.yml
+  publish-release:
+    runs-on: [x64, qemu-host]
+    container:
+      image: ghcr.io/viamrobotics/micro-rdk-dev-env:amd64
+    needs: tests
+    steps:
+    - name : Checkout main branch code
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+    - name: Build esp32 binary
+      run: |
+        bash -c 'export MICRO_RDK_USE_NVS=true && . "$IDF_PATH"/export.sh && . "$ESP_ROOT"/export-esp.sh && make build-esp32-with-cred-bin'
+        cp examples/target/esp32-server-with-cred.bin  micro-rdk-esp32-server.bin
+        sha256sum  micro-rdk-esp32-server.bin  >> sha256sums.txt
+    - name: Check release type
+      id: check-tag
+      run: |
+        if echo ${{ github.event.ref }} | grep -Eq '^refs/tags/v.*rc[0-9]{1}$'; then
+              echo "match=true" >> $GITHUB_OUTPUT
+        else
+              echo "match=false" >> $GITHUB_OUTPUT
+        fi
+    - name: Publish release candidate
+      if: steps.check-tag.outputs.match == 'true'
+      uses: marvinpinto/action-automatic-releases@latest
+      with:
+        repo_token: ${{ secrets.GITHUB_TOKEN }}
+        files: |
+          micro-rdk-esp32-server.bin
+          sha256sums.txt
+        prerelease: true
```

Comment:
Could you dedup this with something like `prerelease: ${{needs.check-tag.outputs.output1}}` per https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs?

I think you would need to make "Publish" its own job though. So something like:

```
jobs:
  tests: ...
  build-release:
      outputs:
        prerelease: ${{ steps.check-tag.outputs.prerelease }}
    steps:
     - name: Checkout...
     - name: Build ...
     - name: Check release type
        id: check-tag
        run: |
          if echo ${{ github.event.ref }} | grep -Eq '^refs/tags/v.*rc[0-9]{1}$'; then
                echo "prerelease=true" >> $GITHUB_OUTPUT
          else
                echo "prerelease=false" >> $GITHUB_OUTPUT
          fi
  publish-release:
    needs: build-release
    steps:
    - publish:
      uses: marvinpinto/action-automatic-releases@latest
      with:
        repo_token: ${{ secrets.GITHUB_TOKEN }}
        files: |
          micro-rdk-esp32-server.bin
          sha256sums.txt
        prerelease: ${{needs.build-release.outputs.prerelease}}
```

Something like that.